### PR TITLE
chore: enable pod security admission

### PR DESCRIPTION
**Pull request by jopit**
**What type of PR is this?**

/kind chore

**What does this PR do / why we need it**:

Updates the code to work with Pod Security Admission, which is enabled by default in k8s 1.24

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**How to test changes / Special notes to the reviewer**:

- Install the operator on a k8s 1.24 system.
- Install an Argo CD instance
```yaml
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: argocd
spec:
  applicationSet: {}
  notifications:
    enabled: true
  ha:
    enabled: false
  grafana:
    enabled: true
  redis:
    disableTLSVerification: false
  dex:
    config: |
      connectors:
      - type: mockCallback
        id: mock
        name: Example
```
- Ensure all workloads are running
```bash
% kubectl get po
NAME                                                READY   STATUS    RESTARTS   AGE
argocd-application-controller-0                     1/1     Running   0          34s
argocd-applicationset-controller-6d99f5bc48-cpbdm   1/1     Running   0          34s
argocd-dex-server-6f4bd74657-jl7zk                  1/1     Running   0          34s
argocd-grafana-7588bd47b5-8xrqt                     1/1     Running   0          34s
argocd-notifications-controller-86ddb6ff44-kqxjk    1/1     Running   0          33s
argocd-redis-7c79b9c5b6-7wmcf                       1/1     Running   0          34s
argocd-repo-server-857cf5d777-j6vnd                 1/1     Running   0          34s
argocd-server-579b655b48-c49rc                      1/1     Running   0          34s
```
- Install the operator on a k8s 1.23 system
- Install an ArgoCD instance
- Ensure all workloads are running

### CODE REVIEW

**Comment by rishabh625** [comment]
Fixes #663

**Comment by rishabh625** [comment]
Fixes #663

File: bundle/manifests/argocd-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -904,6 +904,14 @@ spec:
           - jobs
           verbs:
           - '*'
+        - apiGroups:
```
Comment (rishabh625) [inline bundle/manifests/argocd-operator.clusterserviceversion.yaml:907]: @jopit m not sure if this change is required as following mail of Auth and API , doesn't mentions

**Review by rishabh625 — APPROVED** [review]
other than change in csv , LGTM!!! Thanks @jopit

File: controllers/argocd/deployment.go [hunk 2]
```diff
@@ -457,6 +460,15 @@ func (r *ReconcileArgoCD) reconcileDexDeployment(cr *argoprojv1a1.ArgoCD) error
...
 		Resources: getDexResources(cr),
+		SecurityContext: &corev1.SecurityContext{
+			AllowPrivilegeEscalation: boolPtr(false),
+			Capabilities: &corev1.Capabilities{
+				Drop: []corev1.Capability{
+					"ALL",
+				},
+			},
+			RunAsNonRoot: boolPtr(true),
+		},
```
Comment (ishitasequeira) [inline controllers/argocd/deployment.go:471]: I see that we have added this SecurityContext at multiple places and it looks almost the same. Was wondering if we could move it to a common place and call it everywhere?

**Review by ishitasequeira — APPROVED** [review]
@jopit The overall PR looks good and the testing worked fine. Just added a quick suggestion. Let me know what you think.

**Pull request merged**